addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaBridgeHandler.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.handler;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingStatusInfoBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants;
+import org.openhab.binding.yamahareceiver.discovery.ZoneDiscoveryService;
+import org.openhab.binding.yamahareceiver.internal.protocol.HttpXMLSendReceive;
+import org.openhab.binding.yamahareceiver.internal.protocol.SystemControl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+/**
+ * The {@link YamahaBridgeHandler} is responsible for fetching basic information about the
+ * found AVR and start the zone detection.
+ *
+ * @author David Graeff <david.graeff@web.de>
+ */
+public class YamahaBridgeHandler extends BaseBridgeHandler {
+    private Logger logger = LoggerFactory.getLogger(YamahaBridgeHandler.class);
+    private int refrehInterval = 60; // Default: Every 1min
+    private float relativeVolumeChangeFactor = 0.5f; // Default: 0.5 percent
+    private ScheduledFuture<?> refreshTimer;
+    private ZoneDiscoveryService zoneDiscoveryService;
+
+    private HttpXMLSendReceive xml;
+    private SystemControl.State state = new SystemControl.State();
+
+    public YamahaBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    public float getRelativeVolumeChangeFactor() {
+        return relativeVolumeChangeFactor;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (xml == null || state.host == null) {
+            return;
+        }
+
+        String id = channelUID.getId();
+
+        try {
+            if (command instanceof RefreshType) {
+                refreshFromState(channelUID);
+                return;
+            }
+
+            switch (id) {
```

Comment:
Usually correct, but might be extended in the future. There is still unimplemented functionality. I added a comment.